warning: `QVector<QRect> QRegion::rects() const` is deprecated

```shell
/home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/Plot2DCanvas.cpp: In member function ‘virtual void OpenMS::Plot2DCanvas::paintEvent(QPaintEvent*)’:
/home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/Plot2DCanvas.cpp:1536:45: warning: ‘QVector<QRect> QRegion::rects() const’ is deprecated: Use begin()/end() instead [-Wdeprecated-declarations]
 1536 |     QVector<QRect> rects = e->region().rects();
      |                            ~~~~~~~~~~~~~~~~~^~
In file included from /usr/include/qt/QtGui/qmatrix.h:45,
                 from /usr/include/qt/QtGui/qbrush.h:49,
                 from /usr/include/qt/QtGui/qpalette.h:46,
                 from /usr/include/qt/QtWidgets/qwidget.h:48,
                 from /usr/include/qt/QtWidgets/QWidget:1,
                 from /home/USER/Development/OpenMs/src/openms_gui/include/OpenMS/VISUAL/ColorSelector.h:41,
                 from /home/USER/Development/OpenMs/src/openms_gui/source/VISUAL/Plot2DCanvas.cpp:44:
/usr/include/qt/QtGui/qregion.h:125:20: note: declared here
  125 |     QVector<QRect> rects() const;
      |                    ^~~~~
```

--- a/src/openms_gui/source/VISUAL/Plot2DCanvas.cpp
+++ b/src/openms_gui/source/VISUAL/Plot2DCanvas.cpp
@@ -1533,7 +1533,14 @@ namespace OpenMS
     painter.begin(this);
 
     //copy peak data from buffer
+     /*
+         * Suppressed warning QVector<QRect> QRegion::rects() const is deprecated
+         * Use begin()/end() instead, from Qt 5.8
+         */
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
     QVector<QRect> rects = e->region().rects();
+#pragma GCC diagnostic pop
     for (int i = 0; i < (int)rects.size(); ++i)
     {
       painter.drawImage(rects[i].topLeft(), buffer_, rects[i]);

